src/components/Molecules/Countdown/Countdown.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Text from '../../Atoms/Text/Text';
+import { Wrapper, Digits } from './Countdown.style';
+
+const Countdown = ({
+  endDate, color, endMessage, introMessage
+}) => {
+  const [countdownEnd, setCountdownEnd] = useState(false);
+  const [countdownTime, setCountdownTime] = useState({});
+
+  const countDownDate = new Date(endDate).getTime();
+  // Keep number with two digits
+  const formatNumber = number => (`0${number}`).slice(-2);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const now = new Date().getTime();
+      const distance = countDownDate - now;
+
+      const days = formatNumber(Math.floor(distance / (1000 * 60 * 60 * 24)));
+      const hours = formatNumber(Math.floor((distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60)));
+      const minutes = formatNumber(Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60)));
+      const seconds = formatNumber(Math.floor((distance % (1000 * 60)) / 1000));
+
+      setCountdownTime({
+        days,
+        hours,
+        minutes,
+        seconds
+      });
+
+      if (distance < 0) {
+        setCountdownEnd(true);
+      }
+    }, 1000);
+    return () => clearInterval(interval);
+  }, [countDownDate]);
+
+  const CountDownClock = (
```

Comment:
The loading state doesn't look great ... would it be better to start with '00 : 00 : 00 : 00', or just not render the clock until `useEffect` has executed and set the time?

Reply:
Cool done '00 : 00 : 00 : 00' as initial state. 👍